import-scripts/precache-portals.py

```diff
@@ -0,0 +1,143 @@
+def get_authorization_header(deployment):
+    if deployment not in PROTECTED_PORTALS:
+        return None 
+    if "DATA_ACCESS_TOKEN" not in os.environ:
+        print "No value set for environment variable DATA_ACCESS_TOKEN. Unable to submit requests, exiting..."
+        sys.exit(1)
+    return {"Authorization" : "Bearer %s" % (os.environ("DATA_ACCESS_TOKEN"))}
+
+def get_cancer_studies(cbio_url, authorization_header):
+    cancer_study_ids = []
+    study_api_response = requests.get("%s/api/studies?direction=ASC&pageNumber=0&pageSize=10000000&projection=SUMMARY" % (cbio_url), headers = authorization_header).json()
+    for study in study_api_response:
+        cancer_study_ids.append(study["studyId"])
+    return cancer_study_ids
+
+def cache_fusion_genes(cbio_url, post_body, authorization_header):
+    requests.post("%s/api/fusion-genes/fetch" % (cbio_url), headers = authorization_header, json = post_body)
+
+def cache_mutated_genes(cbio_url, post_body, authorization_header):
+    requests.post("%s/api/mutated-genes/fetch" % (cbio_url), headers = authorization_header,  json = post_body)
+
+def cache_cna_genes(cbio_url, post_body, authorization_header):
+    requests.post("%s/api/cna-genes/fetch" % (cbio_url), headers = authorization_header, json = post_body)
+
+# construct JSON post body for study view endpoints
+# {sampleIdentifiers: [{sampleId: S01, study: study1}, {sampleId: S02, study: study1}...]}
+def construct_sample_list_post_body(cbio_url, cancer_study, authorization_header):
+    samples = []
+    sample_api_response = requests.get("%s/api/studies/%s/samples?direction=ASC&pageNumber=0&pageSize=10000000&projection=SUMMARY" % (cbio_url, cancer_study), headers = authorization_header).json()
+    for sample in sample_api_response:
+        samples.append({"sampleId" : sample["sampleId"], "studyId":sample["studyId"]})
+    post_body = json.loads(json.dumps({"sampleIdentifiers" : samples}))
+    return post_body
+
+def precache_slow_endpoints(cbio_url, post_body, authorization_header):
+    pool = Pool(5)
+    pool.apply_async(cache_fusion_genes, [cbio_url, post_body, authorization_header])
+    pool.apply_async(cache_mutated_genes, [cbio_url, post_body, authorization_header])
+    pool.apply_async(cache_cna_genes, [cbio_url, post_body, authorization_header])
+# ----------------------------------------------------------
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-d", "--deployments", help = "comma seperated list of k8s deplyoments for precaching", required = True)
+    args = parser.parse_args()
+    
+    deployments = map(str.strip, args.deployments.split(','))
+    if not is_valid_deployments(deployments):
+        print "Invalid deployments supplied, exiting..."
+        sys.exit(1)
+
+    # run check and precache for every cbio deployment (e.g genie, genie-private,...)
+    for deployment in deployments:
+        authorization_header = get_authorization_header(deployment)
```

Comment:
I think you only need to get this once since it is the same for all of them right now.